core/arch/arm/mm/core_mmu.c

```diff
@@ -691,15 +693,54 @@ static void dump_mmap_table(struct tee_mmap_region *memory_map)
 	}
 }
 
+static void add_pager_vaspace(struct tee_mmap_region *mmap __maybe_unused,
+			      size_t num_elems __maybe_unused,
+			      vaddr_t begin __maybe_unused,
+			      vaddr_t *end __maybe_unused,
+			      size_t *last __maybe_unused)
+{
+#ifdef CFG_WITH_PAGER
+	size_t size = CFG_TEE_RAM_VA_SIZE - (*end - begin);
+	size_t n;
+	size_t pos = 0;
+
+	if (!size)
+		return;
+
+	if (*last >= (num_elems - 1)) {
+		EMSG("Out of entries (%zu) in memory map", num_elems);
+		panic();
+	}
+
+	for (n = 0; !core_mmap_is_end_of_table(mmap + n); n++)
+		if (map_is_flat_mapped(mmap + n))
+			pos = n + 1;
+
+	assert(pos <= *last);
+	memmove(mmap + pos + 1, mmap + pos,
+		sizeof(struct tee_mmap_region) * (*last - pos));
+	(*last)++;
+	memset(mmap + pos, 0, sizeof(mmap[0]));
+	mmap[pos].type = MEM_AREA_PAGER_VASPACE;
+	mmap[pos].va = *end;
+	mmap[pos].size = size;
+	mmap[pos].region_size = SMALL_PAGE_SIZE;
+	mmap[pos].attr = core_mmu_type_to_attr(MEM_AREA_PAGER_VASPACE);
+
+	*end += size;
```

Comment:
I agree with the edited comment: when pager is disable, `CFG_TEE_RAM_VA_SIZE` is meaningless.
Hmm... maybe usefull also for the kasan ?

However, some platform do define `CFG_TEE_RAM_VA_SIZE` and some piece of code use it. Ok to not condition the sequence here and postponed cleaning of `CFG_TEE_RAM_VA_SIZE` to later if needed.